### PR TITLE
[docs] integrate algolia docsearch, move to sphinx panels

### PR DESCRIPTION
**Pull request by maxpumperla**
Adds search-as-you-type search from Algolia and deprecates a dependency that stood in the way. Here's a preview:

https://ray--21814.org.readthedocs.build/en/21814/

Note that algolia crawls docs.ray.io, hence the search results point to 1.9.2.  This is not a blocking issue for now and can be configured in algolia properly.

### CODE REVIEW

**Comment by maxpumperla** [comment]
@richardliaw let me know what you think!

File: doc/source/cluster/cloud.rst [hunk 1]
```diff
@@ -23,168 +23,167 @@ Ray with cloud providers
 
     /cluster/aws-tips.rst
 
-.. tabs::
-    .. group-tab:: AWS
+.. tabbed::  AWS
```
Comment (richardliaw) [inline doc/source/cluster/cloud.rst:26]: This tabbed stuff is beautiful...

**Comment by richardliaw** [comment]
Screen Shot 2022-01-24 at 9 45 51 AM
Is it possible to make the search centered instead of left-justified?

File: doc/source/cluster/cloud.rst (hunk 1 above)
Comment (maxpumperla) [reply doc/source/cluster/cloud.rst:26]: yeah, much easier!

**Comment by maxpumperla** [comment]
@richardliaw for sure can I center a div with CSS, but that might open Pandora's box. ;) let me do that quickly.

**Comment by maxpumperla** [comment]
@richardliaw done. let's wait a couple of hours for CI, but an approval would be nice!

**Review by richardliaw — APPROVED** [review]
Beautiful https://media.giphy.com/media/QBKagmzOgZnqadpx3B/giphy.gif

**Pull request merged**